index.bs

```diff
@@ -73,34 +73,127 @@ command/response format of WebDriver, this permits events to stream
 from the user agent to the controlling software, better matching the
 evented nature of the browser DOM.
 
+Infrastructure {#infrastructure}
+==========================
+
+This specification depends on the Infra Standard. [[!INFRA]]
+
+Network protocol messages are defined using CDDL. [[!RFC8610]]
+
 Protocol {#protocol}
 ==============
 
 This section defines the basic concepts of the WebDriver BiDi
 protocol. These terms are distinct from their representation at the
 <a href=#transport>transport</a> layer.
 
-This specification uses Concise Data Definition Language (CDDL)
-[[!RFC8610]] to describe the format of messages transmitted between the
-[=local end=] and [=remote end=].
+The protocol is defined using a [[!RFC8610|CDDL]] definition. For the
+convenience of implementors two seperate schema are defined; the <dfn>remote end
+definition</dfn> which defines the format of messages produced on the [=local end=]
+and consumed on the [=remote end=], and the <dfn>local end definition</dfn> which
+defines the format of messages produced on the [=remote end=] and consumed on
+the [=local end=]
+
+## Definition ## {#protocol-definition}
+
+Issue: Should this be an appendix?
+
+This section gives the initial contents of the [=remote end definition=] and
+[=local end definition=]. These are augmented by the definition fragments defined in
+the remainder of the specification.
+
+[=Remote end definition=]
+
+```
+Command = {
+  id: uint,
+  CommandData,
+}
+
+CommandData = (
+  SessionCommand
+)
+
+EmptyParams = { *text }
+```
+
+[=Local end definition=]
+
+```
+Message = (
+  CommandResponse //
+  Event
+)
 
-Note: These definitions do not form any normative requirements, rather
-they are used elsewhere as part of such requirements.
+CommandResponse = (
+  id: uint,
+  ResponseData,
+)
+
+ResponseData = (
+  Error //
+  CommandResult
+)
+
+Error = (
+  id: uint,
```

Comment:
Nit: Looks like `id` was moved to CommandResponse but not removed from Error here.